File: gurri/0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        maxArea = 0

        while (l < r):
            # 갱신
            maxArea = max(maxArea, (r - l)* min(
                height[l], height[r]
            ))

            # 이동
            if height[l] < height[r] : l+= 1
            else : r -= 1

        return maxArea

File: gurri/0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        // 두개의 선을 선택 - 가장 많은 물을 담을 수 있는 용기를 찾기
        // 용기의 너비 - 두선 사이의 거리 / 높이 - 낮은 선의 높이

        int maxArea = 0;
        int left =0;
        int right = height.size() - 1;

        while (left < right){
            maxArea = max(maxArea, (right - left) * min(height[left], height[right]));

            if (height[left] < height[right]){
                left++;
            } else{
                right --;
            }


        }      
        return maxArea;
    }
};